pkg/scheduler/core/generic_scheduler.go

```diff
@@ -434,21 +441,57 @@ func (g *genericScheduler) prioritizeNodes(
 		return nil, scoreStatus.AsError()
 	}
 
-	if klog.V(10).Enabled() {
-		for plugin, nodeScoreList := range scoresMap {
-			for _, nodeScore := range nodeScoreList {
-				klog.InfoS("Plugin scored node for pod", "pod", klog.KObj(pod), "plugin", plugin, "node", nodeScore.Name, "score", nodeScore.Score)
-			}
-		}
-	}
-
 	// Summarize all scores.
 	result := make(framework.NodeScoreList, 0, len(nodes))
+	totalPluginScores := make(map[string]int64)
 
 	for i := range nodes {
 		result = append(result, framework.NodeScore{Name: nodes[i].Name, Score: 0})
 		for j := range scoresMap {
 			result[i].Score += scoresMap[j][i].Score
+			totalPluginScores[j] += scoresMap[j][i].Score
+		}
+	}
+
+	// Enhanced plugin score logging. The goal of this block is to show the highest scoring plugins on each node,
+	// and the average score for those plugins across all nodes.
```

Comment:
I want to argue that provide such information is not always helpful for some cases, lower scoring plugins maybe the culprit for why scheduler doesn't choose the node.

Reply:
I agree, there are some cases where seeing the lowest-scoring plugins will be more helpful. I chose to go with the top scoring plugins because there are a lot of cases where someone asks, "why didn't the pod go on X node when I have this plugin enabled?" and the answer is because another plugin outscored their desired plugin (usually balanced resource allocation). On the flip side, that also means their plugin scored low. So maybe there is some way to show both?

Reply:
I think top scores is almost usually enough. At least if their plugin of interest is missing they can figure that it's scoring low. And if they really want to know everything, they can enable V(10)

Reply:
> And if they really want to know everything, they can enable V(10)

That's a good point too